pkg/apis/pipeline/v1alpha1/pipeline_types.go

```diff
@@ -59,6 +59,19 @@ type Pipeline struct {
 	// Spec holds the desired state of the Pipeline from the client
 	// +optional
 	Spec PipelineSpec `json:"spec"`
+
+	// Status is deprecated.
+	// It usually is used to communicate the observed state of the Pipeline from
+	// the controller, but was unused as there is no controller for Pipeline.
+	// +optional
+	Status PipelineStatus `json:"status,omitempty"`
```

Comment:
`omitempty` has no effect on struct fields that aren't pointers. This will result in clients always sending `"status":{}`, so we'll never be able to upgrade servers to stop accepting it.

If we change this to `*PipelineStatus` then new clients won't send it, but servers will at least still know about it and will be able to decode it.

Then in the future we can remove the field once we believe clients have upgraded to not rely on it.

Reply:
@ImJasonH good point… forgot to modify this to a pointer… :upside_down_face:

Reply:
My goal is to not have this in `v1alpha2` (aka `v1beta1`), that way, it fades away once we remove `v1alpha1` :angel: